backend/pequod/services/pathfinding_service.py

```
@@ -0,0 +1,72 @@
+                    end_node_data = G.nodes[v]
+                    path_coordinates.append({'lat': end_node_data['y'], 'lon': end_node_data['x']})
+
+        return {
+            'start_node_osmid': start_node_osmid,
+            'end_node_osmid': end_node_osmid,
+            'path_coordinates': path_coordinates,
+            'total_length_meters': round(path_length_meters, 2)
+        }
+
+    except nx.NetworkXNoPath:
+        raise nx.NetworkXNoPath(f'No path found between the nearest nodes ({start_node_osmid} and {end_node_osmid}). The locations might be in disconnected parts of the graph.')
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)